.github/workflows/test.yaml

```diff
@@ -19,12 +19,14 @@ jobs:
             
       - run: flutter --version
       - run: flutter pub get
+      - run: flutter pub get --directory=example/viam_example_app
 
+      # Verify formatting in all of example, test & lib except lib/gen/
       - name: Verify formatting
-        run: dart format --output=none .
+        run: dart format --set-exit-if-changed example test lib/src/components lib/src/di lib/src/domain lib/src/media lib/src/proto lib/src/resource lib/src/robot lib/src/rpc
```

Comment:
This seems like an inclusionary selection. It would be nice if it were exclusionary (just _ignored_ `lib/gen`). That way, we don't have to remember to add to this when new directories are added.

Reply:
I agree, I don't think this is a good way to do it and i'd prefer to just exclude the one folder. I will look into that again 👀

Reply:
per offline discussion, if there's not a way to do this I wonder if we could do something like `dart format --set-exit-if-changed $(dart listpackages ./ | grep -v lib/gen)` where `list packages` is some command that may or may not exist to list all dart packages. no worries if not.

Reply:
dart does not have a convenient way to get that list of files, i could write my own script, but that feels heavy handed. I would probably prefer to move the location of `gen/` before that.

Reply:
Use `find`: `dart format --line-length=140 $(find ./lib -name "*.dart" -not -path "./lib/src/gen/*")` will work on all files in `lib`, excluding files in `gen`

Reply:
great callout, that cleaned this up nicely. Thank you!!